Mediapipe.Net/Framework/CalculatorGraph.cs

```diff
@@ -0,0 +1,217 @@
+
+        public CalculatorGraph() : base()
+        {
+            UnsafeNativeMethods.mp_CalculatorGraph__(out var ptr).Assert();
+            Ptr = ptr;
+        }
+
+        public CalculatorGraph(string textFormatConfig) : base()
+        {
+            UnsafeNativeMethods.mp_CalculatorGraph__PKc(textFormatConfig, out var ptr).Assert();
+            Ptr = ptr;
+        }
+
+        public CalculatorGraph(byte[] serializedConfig) : base()
+        {
+            UnsafeNativeMethods.mp_CalculatorGraph__PKc_i(serializedConfig, serializedConfig.Length, out var ptr).Assert();
+            Ptr = ptr;
+        }
+
+        public CalculatorGraph(CalculatorGraphConfig config) : this(config.ToByteArray()) { }
+
+        protected override void DeleteMpPtr() => UnsafeNativeMethods.mp_CalculatorGraph__delete(Ptr);
+
+        public Status Initialize(CalculatorGraphConfig config)
+        {
+            var bytes = config.ToByteArray();
+            UnsafeNativeMethods.mp_CalculatorGraph__Initialize__PKc_i(MpPtr, bytes, bytes.Length, out var statusPtr).Assert();
+
+            GC.KeepAlive(this);
+            return new Status(statusPtr);
+        }
+
+        public Status Initialize(CalculatorGraphConfig config, SidePacket sidePacket)
+        {
+            var bytes = config.ToByteArray();
+            UnsafeNativeMethods.mp_CalculatorGraph__Initialize__PKc_i_Rsp(MpPtr, bytes, bytes.Length, sidePacket.MpPtr, out var statusPtr).Assert();
+
+            GC.KeepAlive(this);
+            return new Status(statusPtr);
+        }
+
+        /// <remarks>Crashes if config is not set</remarks>
+        public CalculatorGraphConfig Config()
+        {
+            UnsafeNativeMethods.mp_CalculatorGraph__Config(MpPtr, out var serializedProto).Assert();
+            GC.KeepAlive(this);
+
+            var config = serializedProto.Deserialize(CalculatorGraphConfig.Parser);
+            serializedProto.Dispose();
+
+            return config;
+        }
+
+        public Status ObserveOutputStream(string streamName, NativePacketCallback nativePacketCallback, bool observeTimestampBounds = false)
+        {
+            UnsafeNativeMethods.mp_CalculatorGraph__ObserveOutputStream__PKc_PF_b(MpPtr, streamName, nativePacketCallback, observeTimestampBounds, out var statusPtr).Assert();
+
+            GC.KeepAlive(this);
+            return new Status(statusPtr);
+        }
+
+        public Status ObserveOutputStream<TPacket, TValue>(string streamName, PacketCallback<TPacket, TValue> packetCallback, bool observeTimestampBounds, out GCHandle callbackHandle) where TPacket : Packet<TValue>
+        {
+            NativePacketCallback nativePacketCallback = (_, packetPtr) =>
+            {
+                Status status;
+                try
+                {
+                    var packet = (TPacket?)Activator.CreateInstance(typeof(TPacket), packetPtr, false);
```

Comment:
yeah i think we should have a bool here at the start of the method that is false by default (useReflection)
if false it'll use a switch to decide the packet type (so only builtin but faster (itll have to be measured)) and if true itll keep using reflection

Reply:
That's something to investigate later, after everything including tests has been ported one-to-one